internal/providers/nscloud/api/types.go

```diff
@@ -81,6 +87,46 @@ type KubernetesCluster struct {
 	Label []*LabelEntry `json:"label,omitempty"`
 }
 
+type TailLogsRequest struct {
+	ClusterID string        `json:"cluster_id,omitempty"`
+	StartTs   time.Time     `json:"start_ts,omitempty"`
+	Selector  *LogsSelector `json:"selector,omitempty"`
+}
+
+type GetLogsRequest struct {
+	ClusterID string        `json:"cluster_id,omitempty"`
+	StartTs   time.Time     `json:"start_ts,omitempty"`
+	EndTs     time.Time     `json:"end_ts,omitempty"`
+	Selector  *LogsSelector `json:"selector,omitempty"`
+}
+
+type GetLogsResponse struct {
+	LogBlock []LogBlock `json:"log_block,omitempty"`
+}
+
+type LogsSelector struct {
+	Namespace string `json:"namespace,omitempty"`
+	Pod       string `json:"pod,omitempty"`
+	Container string `json:"container,omitempty"`
+}
+
+type LogBlock struct {
+	Namespace string    `json:"namespace,omitempty"`
+	Pod       string    `json:"pod,omitempty"`
+	Container string    `json:"container,omitempty"`
+	Line      []LogLine `json:"line,omitempty"`
+}
+
+type LogLine struct {
+	Timestamp time.Time `json:"timestamp,omitempty"`
+	Content   string    `json:"content,omitempty"`
+	Stream    string    `json:"stream,omitempty"`
+}
+
+func (l LogLine) Fprint(w io.Writer) {
```

Comment:
I'd find it more natural to offer `.String()`

Reply:
done